x/staking/client/cli/query.go

```diff
@@ -564,40 +611,48 @@ $ %s query staking redelegation cosmos1gghjut3ccd8ay0zduzj64hwre2fxs9ld75ru9p
 			),
 		),
 		RunE: func(cmd *cobra.Command, args []string) error {
-			clientCtx := client.NewContext().WithCodec(cdc).WithJSONMarshaler(cdc)
-
-			delAddr, err := sdk.AccAddressFromBech32(args[0])
+			clientCtx := client.GetClientContextFromCmd(cmd)
+			clientCtx, err := client.ReadQueryCommandFlags(clientCtx, cmd.Flags())
 			if err != nil {
 				return err
 			}
 
-			bz, err := cdc.MarshalJSON(types.QueryRedelegationParams{DelegatorAddr: delAddr})
+			queryClient := types.NewQueryClient(clientCtx)
+
+			delAddr, err := sdk.AccAddressFromBech32(args[0])
 			if err != nil {
 				return err
 			}
 
-			route := fmt.Sprintf("custom/%s/%s", queryRoute, types.QueryRedelegations)
-			res, _, err := clientCtx.QueryWithData(route, bz)
+			pageReq, err := client.ReadPageRequest(cmd.Flags())
 			if err != nil {
 				return err
 			}
 
-			var resp types.RedelegationResponses
-			if err := cdc.UnmarshalJSON(res, &resp); err != nil {
+			params := &types.QueryRedelegationsRequest{
+				DelegatorAddr: delAddr,
+				Req:           pageReq,
+			}
+
+			res, err := queryClient.Redelegations(context.Background(), params)
+			if err != nil {
 				return err
 			}
 
-			return clientCtx.PrintOutput(resp)
+			return clientCtx.PrintOutput(res.RedelegationResponses)
 		},
 	}
-
+	cmd.Flags().String(flags.FlagPageKey, "", "pagination page-key of delegator redelegations to query for")
+	cmd.Flags().Uint64(flags.FlagOffset, 1, "pagination offset of delegator redelegations to query for")
+	cmd.Flags().Uint64(flags.FlagLimit, 100, "pagination limit of delegator redelegations to query for")
+	cmd.Flags().Bool(flags.FlagCountTotal, false, "total number of records in delegator redelegations to query for")
```

Comment:
I feel like these 4 lines could also be factored into a helper function `flags.AddPaginationFlagsToCmd`